src/sql/command.py

```diff
@@ -89,3 +88,30 @@ def connection(self):
     def result_var(self):
         """Returns the result_var"""
         return self.parsed["result_var"]
+
+    def _var_expand(self, magic, user_ns, line, cell):
+        """
+        Support for the variable substition in the SQL clause
+        For now, we have enabled two ways:
+        1. Latest format, {{a}}, we use jinja2 to parse the string with {{a}} format
+        2. Legacy format, {a}, $a, and :a format.
+
+        We will deprecate the legacy format feature in next major version
+        """
+        self.is_legacy_var_expand_parsed = False
+        # Latest format parsing
+        # TODO: support --param and --use-global logic here
+        # Ref: https://github.com/ploomber/jupysql/issues/93
+        line = Template(line).render(user_ns)
+        cell = Template(cell).render(user_ns)
+        # Legacy format parsing
+        parsed_cell = magic.shell.var_expand(cell, depth=2)
+        parsed_line = magic.shell.var_expand(line, depth=2)
```

Comment:
what is the depth parameter doing?

Reply:
I checked [the doc again](https://ipython.readthedocs.io/en/stable/api/generated/IPython.core.interactiveshell.html#IPython.core.interactiveshell.InteractiveShell.var_expand) and it's for indicates how many frames above the caller should be walked to look for the local namespace where to expand variables.

[The source code of iPython](https://github.com/ipython/ipython/blob/main/IPython/core/interactiveshell.py#L2361) is setting as depth = 1, we should use the same

Reply:
Those test cases are covering
`test_variable_substitution_legacy_warning_message_single_curly`
`test_variable_substitution_legacy_warning_message_colon`

Reply:
ok, let's keep 2